Enrich security vulnerabilities using Port AI (#2746)
port-labs/port-docs: docs/guides/all/enrich-security-vulnerability-using-ai.md

            "format": "markdown"
          }
    ```
5. Click `Save`

### Creating a new blueprint

> If you do not have a security integration installed, you will need to create a new blueprint. Follow these steps:

Reviewer: @PeyGis does it make sense to manage/monitor vulnerabilities without an integration?

Reviewer: Meaning, maybe it can be a prerequisite? to have either an integration or a blueprint that stores vulnerabilities

  "icon": "AI",
  "trigger": {
    "type": "automation",
    "event": {
      "type": "ENTITY_UPDATED",
      "blueprintIdentifier": "securityVulnerability"
    },
    "condition": {
      "type": "JQ",
      "expressions": [
        ".diff.before.properties.severity != \"critical\"",
        ".diff.after.properties.severity == \"critical\""
      ],
      "combinator": "and"
    }
  },
  "invocationMethod": {
    "type": "WEBHOOK",
    "url": "https://api.getport.io/v1/agent/vulnerability_enrichment_agent/invoke",
    "agent": false,
    "synchronized": true,
    "method": "POST",
    "headers": {
      "RUN_ID": "{{ .run.id }}",
      "Content-Type": "application/json"
    },
    "body": {
      "prompt": "Generate a summary and remediation steps for vulnerability with entity identifier {{ .event.context.entityIdentifier }}. The issue has been updated with the following details: {{ .event.diff.after.properties }}",
      "labels": {
        "source": "Security AI Summary",
        "entityIdentifier": "{{ .event.context.entityIdentifier }}"
      }
    }
  },
  "publish": true
}
```
</details>

4. Click `Create` to save the automation

> ### Automation 2: Trigger Claude Code to fix vulnerability

Reviewer: @PeyGis this is just a small comment, but we rely on Claude Code in all guides almost. I suggest we'll make it more dynamic, for example use Gemini here or Coopilot.
Just an idea

  "icon": "Vulnerability",
  "trigger": {
    "type": "automation",
    "event": {
      "type": "ENTITY_UPDATED",
      "blueprintIdentifier": "securityVulnerability"
    },
    "condition": {
      "type": "JQ",
      "expressions": [
        ".diff.before.properties.ai_summary == null",
        ".diff.after.properties.ai_summary != null"
      ],
      "combinator": "and"
    }
  },
  "invocationMethod": {
    "type": "WEBHOOK",
    "url": "https://api.getport.io/v1/actions/run_claude_code/runs",
    "agent": false,
    "synchronized": false,
    "method": "POST",
    "headers": {
      "RUN_ID": "{{ .run.id }}",
      "Content-Type": "application/json"
    },
    "body": {
      "properties": {
        "service": "{{ .event.diff.after.relations.service}}",
        "prompt": "Here is the information about the security vulnerability: {{ .event.diff.after }}.\n\nGenerate a code fix for the issue based on the AI-provided summary {{ .event.diff.after.properties.ai_summary }}.\nAfter generating the code, open a PR with a description summarizing what was fixed and why."
      }
    }
  },
  "publish": true
}
```
</details>

4. Click `Create` to save the automation

> ## Test your workflow

Reviewer: If you have a screenshot showing something here, like a dashbord or a PR, would be great

  "trigger": {
    "type": "self-service",
    "operation": "DAY-2",
    "userInputs": {
      "properties": {
        "ai_summary": {
          "type": "string",
          "title": "AI Summary",
          "format": "markdown"
        }
      },
      "required": ["ai_summary"],
      "order": ["ai_summary"]
    },
    "blueprintIdentifier": "securityVulnerability"
  },
  "invocationMethod": {
    "type": "WEBHOOK",
    "url": "https://api.getport.io/v1/blueprints/{{.action.blueprint}}/entities/{{.entity.identifier}}",
    "agent": false,
    "synchronized": true,
    "method": "PATCH",
    "headers": {
      "RUN_ID": "{{ .run.id }}",
      "Content-Type": "application/json"
    },
    "body": {
      "properties": {
        "ai_summary": "{{ .inputs.ai_summary }}"
      }
    }
  },
  "requiredApproval": false
}
```
</details>

:::tip Configuration adjustments
Make sure to adjust the following fields:
- **`blueprintIdentifier`** - Change this to match the blueprint representing security issues in your data model (e.g., `snykVulnerability`, `wizIssue`, `sonarQubeIssue` for Snyk, Wiz, and SonarQube respectively)
:::

5. Click `Save`

## Create your AI agent

> Now you will create an AI agent that analyzes vulnerabilities and calls the action above to update them with AI-generated summaries.

Reviewer 2: ```suggestion
Now let's create an AI agent that analyzes vulnerabilities and calls the action above to update them with AI-generated summaries.
```

    "status": "active",
    "allowed_blueprints": [
      "securityVulnerability",
      "githubRepo"
    ],
    "allowed_actions": [
      "update_vulnerability_with_ai_summary"
    ],
    "prompt": "You are an agent responsible for analyzing security vulnerabilities and providing actionable recommendations.\n\n### Guidelines\n- Summarize the vulnerability in simple terms.\n- Explain the potential impact if exploited.\n- Recommend step-by-step remediation strategies.\n- Suggest best practices to prevent similar vulnerabilities.\n- Tailor responses to the severity (Critical, High, Medium, Low).\n\nAfter generating the summary, call the \"update_vulnerability_with_ai_summary\" action to update the vulnerability entity with the summary",
    "execution_mode": "Automatic",
    "conversation_starters": [
      "Explain the impact of CVE-2025-1234",
      "Suggest a remediation plan for this SQL injection vulnerability",
      "What steps should I take to patch a high severity XSS issue?"
    ]
  },
  "relations": {}
}
```
</details>

5. Click `Create` to save the agent

## Set up automations

> You will now create two automations that orchestrate the complete vulnerability enrichment and remediation workflow:

Reviewer 2: ```suggestion
We will now create two automations that orchestrate the complete vulnerability enrichment and remediation workflow:
```
